flutter-idea/src/io/flutter/vmService/VmServiceWrapper.java

```diff
@@ -473,45 +485,105 @@ public void addBreakpoint(@NotNull final String isolateId,
+            final String category = "breakpoint";
+
+            // For internal bazel projects, report files where mapping failed.
+            if (WorkspaceCache.getInstance(myDebugProcess.getSession().getProject()).isBazel()) {
+              if (resolvedUri.contains("google3")) {
+                System.out.println(String.format("no-package-uri|%s", resolvedUri));
+                analytics.sendEvent(category, String.format("no-package-uri|%s", resolvedUri));
+              }
+            }
 
-            checkDone();
+            consumer.received(breakpointResponses, errorResponses);
+            return;
           }
 
-          @Override
-          public void received(Sentinel response) {
-            checkDone();
-          }
+          final String scriptUri = uris.get(0);
 
-          @Override
-          public void onError(RPCError error) {
-            errorResponses.add(error);
+          myVmService.addBreakpointWithScriptUri(isolateId, scriptUri, line, new AddBreakpointWithScriptUriConsumer() {
+            @Override
+            public void received(Breakpoint response) {
+              breakpointResponses.add(response);
+              breakpointNumbersToCanonicalMap.put(response.getBreakpointNumber(), canonicalBreakpoint);
 
-            checkDone();
-          }
+              checkDone();
+            }
+
+            @Override
+            public void received(Sentinel response) {
+              checkDone();
+            }
+
+            @Override
+            public void onError(RPCError error) {
+              errorResponses.add(error);
+
+              checkDone();
+            }
 
-          private void checkDone() {
-            if (scriptUris.size() == breakpointResponses.size() + errorResponses.size()) {
+            private void checkDone() {
               consumer.received(breakpointResponses, errorResponses);
             }
-          }
-        });
-      }
+          });
+        }
+
+        @Override
+        public void onError(RPCError error) {
+          errorResponses.add(error);
+          consumer.received(breakpointResponses, errorResponses);
+        }
+      });
     });
   }
 
+  private String getResolvedUri(XSourcePosition position) {
+    final String url = position.getFile().getUrl();
+
+    if (WorkspaceCache.getInstance(myDebugProcess.getSession().getProject()).isBazel()) {
+      final String root = "google3";
+
+      // Look for a generated file path.
+      final String genFilePattern = root + "/blaze-.*?/(.*)";
+      final Pattern pattern = Pattern.compile(genFilePattern);
+      final Matcher matcher = pattern.matcher(url);
+      if (matcher.find()) {
+        final String path = matcher.group(1);
+        return root + ":///" + path;
+      }
+
+      // Look for root.
+      final int rootIdx = url.indexOf(root);
+      if (rootIdx > 0) {
+        return root + ":///" + url.substring(rootIdx + root.length() + 1);
```

Comment:
nit: just make this "google3:///" as the fact that this google3 uri scheme happens to make the google3 root directory is somewhat arbitrary.